soroban-sdk/src/map.rs

```diff
@@ -428,30 +429,36 @@ where
     #[inline(always)]
     pub fn iter_unchecked(&self) -> UncheckedIter<MapIter<K, V>, (K, V), ConversionError>
     where
-        K: IntoVal<Env, RawVal> + TryFromVal<Env, RawVal> + Clone,
-        K::Error: Debug,
-        V: IntoVal<Env, RawVal> + TryFromVal<Env, RawVal> + Clone,
-        V::Error: Debug,
+        K: IntoVal<Env, RawVal> + Clone,
+        <RawVal as TryIntoVal<Env, K>>::Error: Debug,
```

Comment:
I don't like how every "T::Error" now have to be expressed as
"<RawVal as TryIntoVal<Env, K>>::Error", not sure if Rust allows a better way to define this type for an entire struct.

Reply:
Need something like the [inherent associated type](https://github.com/rust-lang/rust/issues/8995).